apps/condo/domains/notification/adapters/redStoreAdapter.js

```diff
@@ -0,0 +1,265 @@
+            responses: [],
+            successCount: 0,
+            failureCount: 0,
+        }
+    }
+
+    /**
+     * Mimics redStore failure response
+     * @returns {{success: boolean, error: {errorInfo: {code: string, message: string}}}}
+     */
+    static getFakeErrorResponse () {
+        return {
+            success: false,
+            type: 'Fake',
+            error: {
+                errorInfo: {
+                    code: 'fake-error',
+                    message: 'Fake error message',
+                    status: 403,
+                },
+            },
+        }
+    }
+
+    /**
+     * Mimics redStore success response
+     * @returns {{success: boolean, messageId: string}}
+     */
+    static getFakeSuccessResponse () {
+        return {
+            success: true,
+            type: 'Fake',
+            messageId: `fake-success-message/${faker.datatype.uuid()}`,
+        }
+    }
+
+    /**
+     * For testing purpose we have to emulate redStore response for predefined FAKE tokens,
+     * because it's almost impossible to get real redStore push token in automated way.
+     * @param result
+     * @param fakeNotifications
+     * @returns {*}
+     */
+    static injectFakeResults (result, fakeNotifications) {
+        const mixed = !isObject(result) || isEmpty(result) ? RedStoreAdapter.getEmptyResult() : JSON.parse(JSON.stringify(result))
+
+        fakeNotifications.forEach(({ token }) => {
+            if (token.startsWith(PUSH_FAKE_TOKEN_SUCCESS)) {
+                mixed.successCount++
+                mixed.responses.push(RedStoreAdapter.getFakeSuccessResponse())
+            }
+
+            if (token.startsWith(PUSH_FAKE_TOKEN_FAIL)) {
+                mixed.failureCount++
+                mixed.responses.push(RedStoreAdapter.getFakeErrorResponse())
+            }
+        })
+
+        return mixed
+    }
+
+    /**
+     * Manages to send notification to all available pushTokens of the user.
+     * Also supports PUSH_FAKE_TOKEN_SUCCESS and PUSH_FAKE_TOKEN_FAIL for testing purposes
+     * Would try to send request to redStore only if redStore is initialized and `tokens` array contains real (non-fake) items.
+     * Would succeed if at least one real token succeeds in delivering notification through redStore, or
+     * PUSH_FAKE_TOKEN_SUCCESS provided within tokens
+     * @param notification
+     * @param tokens
+     * @param data
+     * @param pushTypes
+     * @returns {Promise<null|(boolean|T|{state: string, error: *})[]>}
+     */
+    async sendNotification ({ notification, data, tokens, pushTypes } = {}, isVoIP = false) {
+        if (!tokens || isEmpty(tokens)) return [false, { error: 'No pushTokens available.' }]
+
+        const [notifications, fakeNotifications, pushContext] = RedStoreAdapter.prepareBatchData(notification, data, tokens, pushTypes, isVoIP)
+        let result
+
+        // If we come up to here and no real tokens provided, that means fakeNotifications contains
```

Comment:
It looks like you trying to hide a real world error. You must throw the error in that case. Don't try to guess